java/java.source.base/src/org/netbeans/api/java/source/SourceUtils.java

```diff
@@ -684,6 +688,9 @@ private abstract static class Match {
         }
 
         final boolean apply(final FileObject fo) {
+            if (fo.isFolder()) {
+                return false;
```

Comment:
The key fix.

Reply:
OK with me, although I wonder if checking for `isJava()` first might be the right way as this partly overlaps?  Could this whole method just be ...

```java
return isJava(fo) && match(fo.getName(), name);
```

Reply:
Well that is what was added in #5152, presumably necessary for that fix.

Reply:
Ah, missed that fact!  Only a concern if there's a file without extension that matches then?  Probably less likely.

Reply:
I suppose so. My limited understanding of the original change was that `name` could sometimes include the extension, like `X.scala` or something? At any rate, clearly it was intended to match on files and never folders.